### ISSUE TITLE
Migrating wiki links on Github

### ISSUE THREAD

**Issue by Lassulus** [NixOS/nixos-wiki-infra#105]
tracking issue for migration of github links from nixos.wiki to wiki.nixos.org

**Comment by montchr** [comment]
Hey, thank you for the pull request.  There were a couple issues with suggested replacements that would have ended up as 404 errors:

[Update NixOS Wiki link by nixos-wiki · Pull Request #114 · montchr/dotfield](https://github.com/montchr/dotfield/pull/114#discussion_r1658054291)

> -   The official wiki only provides a low-resolution `favicon.ico`, no `favicon.png`: [[wiki.nixos.org/favicon.ico](https://wiki.nixos.org/favicon.ico)](https://wiki.nixos.org/favicon.ico)
> -   The official wiki requires the `w/` subpath/prefix before `index.php`, which affects search URLs and possibly other things
> 
> It might be helpful for future PRs to consider mappings for these particular URLs, as other users likely have configured a similar custom Firefox search engine that needs updating. But I can understand if that's out of scope or more effort than it's worth. The important part is bringing attention to the official wiki, so I still appreciate that 🙂 (and was pleasantly surprised to see this PR just after I had prepared my own fixes committed in [https://github.com/montchr/dotfield/commit/91362abce1e11e184e6bd7b6b5e489b92c35eeef](https://github.com/montchr/dotfield/commit/91362abce1e11e184e6bd7b6b5e489b92c35eeef)).

**Comment by lordkekz** [comment]
In my repo you created a second PR, based on the out-of-date fork that was created for the first one. I already merged the first one so there is no need for another: https://github.com/lordkekz/dotfiles/pull/2
Mentioning this because it might affect other repos as well.

**Comment by Mic92** [comment]
@lordkekz and the PR was still open while created the second one? I thought there was some detection if a pull request is still open. But maybe it couldn't catch this one.

**Comment by lordkekz** [comment]
> @lordkekz and the PR was still open while created the second one? I thought there was some detection if a pull request is still open. But maybe it couldn't catch this one.

No, I had already merged the first PR. But it shouldn't send a new MR in my case, because in `main` everything is migrated already.

To avoid this case, it could sync the fork to the target repo's main branch, and only then check if there are any remaining changes.

**Comment by Mic92** [comment]
@lordkekz ok, but this link is on the main branch: https://github.com/lordkekz/dotfiles/blob/89cb931ec2b750edd59bb9a91246f5998cb329ba/overlays/default.nix#L12, no?

**Comment by lordkekz** [comment]
@Mic92
Yes, that's on my main branch, but it already points to the correct (new) path, or am I missing something?
Right now, my repo's main branch does not contain any links to `nixos.wiki` because they have already been replaced in the first PR.

In lordkekz/dotfiles#2 the program is trying to do the same change (again replace `nixos.wiki` with `wiki.nixos.org`) which has already been applied in the current main:
https://github.com/lordkekz/dotfiles/pull/2/files#diff-814a8207fcd4836674805ac0c80b79b23092f6cdba873113c04ffdf861d3ac61R12

I think this is what is happening:
1. For the first PR, it forked my repo: https://github.com/wiki-nixos/cloned-repo-70ebcd46-0aff-4a40-8d83-b144e01b3661
2. Then, it sent the first PR, you offered some suggestions and I merged that MR.
3. This week, it reused the same fork, without syncing it. That means that the fork's main in which the program searches for links to the old wiki and from which it branches off, is now out of date. Crucially, it doesn't contain the commits that were added from the first MR.
   It says "This branch is [13 commits behind](https://github.com/wiki-nixos/cloned-repo-70ebcd46-0aff-4a40-8d83-b144e01b3661/compare/main...lordkekz%3Adotfiles%3Amain) lordkekz/dotfiles:main." when you look at the fork's main: https://github.com/wiki-nixos/cloned-repo-70ebcd46-0aff-4a40-8d83-b144e01b3661
5. Then the program created the PR based on the outdated branch, and of course found the same old links again.

I can't seem to find a way to show git log graphs including PRs? I feel like GitHub makes it hard to visualize this problem.

**Comment by Mic92** [comment]
Ah. sorry. I misread the domain as well for some reason.

**Comment by lordkekz** [comment]
Ah, no worries
Thank you for your work btw :)
